src/Ormolu/Processing/Preprocess.hs

```diff
@@ -4,126 +4,71 @@
 
 -- | Preprocessing for input source code.
 module Ormolu.Processing.Preprocess
-  ( preprocess,
+  ( linesNotToFormat,
   )
 where
 
 import Control.Monad
 import Data.Char (isSpace)
+import Data.IntSet (IntSet)
+import qualified Data.IntSet as IntSet
 import qualified Data.List as L
-import Data.Maybe (isJust, maybeToList)
-import GHC.Data.FastString
-import GHC.Types.SrcLoc
+import Data.Maybe (isJust)
 import Ormolu.Config (RegionDeltas (..))
-import Ormolu.Parser.Shebang (isShebang)
 import Ormolu.Processing.Common
-import qualified Ormolu.Processing.Cpp as Cpp
+import Ormolu.Processing.Cpp
 
--- | Transform given input possibly returning comments extracted from it.
--- This handles LINE pragmas, CPP, shebangs, and the magic comments for
--- enabling\/disabling of Ormolu.
-preprocess ::
-  -- | File name, just to use in the spans
-  FilePath ->
-  -- | Input to process
-  String ->
-  -- | Region deltas
-  RegionDeltas ->
-  -- | Literal prefix, pre-processed input, literal suffix, extra comments
-  (String, String, String, [RealLocated String])
-preprocess path input RegionDeltas {..} =
-  go 1 OrmoluEnabled Cpp.Outside id id regionLines
+-- | All lines we are not supposed to format.
+linesNotToFormat :: RegionDeltas -> String -> IntSet
+linesNotToFormat region@RegionDeltas {..} input = unconsideredLines <> disabledLines
```

Comment:
Can you try to insert some newlines so that no line is longer than 80 characters?